src/sql/base/query/browser/untitledQueryEditorInput.ts

```diff
@@ -44,6 +48,26 @@ export class UntitledQueryEditorInput extends QueryEditorInput implements IUntit
 		return this.text.model.hasAssociatedFilePath;
 	}
 
+	override async save(group: GroupIdentifier, options?: ISaveOptions): Promise<IEditorInput | undefined> {
+		let preProcessed = await this.text.save(group, options);
+		let newUri = preProcessed.resource.toString(true);
+		this._results.uri = newUri;
+		await this.notifyConnectionUriChanged(newUri);
+		let newInput = this.instantiationService.createInstance(FileQueryEditorInput, '', (preProcessed as FileEditorInput), this.results);
+		newInput.state.setState(this.state);
+		return newInput;
+	}
+
+	override async saveAs(group: GroupIdentifier, options?: ISaveOptions): Promise<IEditorInput | undefined> {
+		let preProcessed = await this.text.saveAs(group, options);
+		let newUri = preProcessed.resource.toString(true);
```

Comment:
These last lines are duplicated between the two save functions, would be nice to combine them.